modules/architecture-machine-config-pools.adoc

```diff
@@ -0,0 +1,24 @@
+// Module included in the following assemblies:
+//
+// * architecture/control-plane.adoc
+
+[id="architecture-machine-config-pools_{context}"]
+= Node configuration management with machine config pools
+
+Machines that manage the cluster and its workloads are divided into groups based on the types of resources they handle. These groups of machines are called machine config pools (MCP). Each MCP manages a set of nodes and its corresponding machine configs. The role of the node determines which MCP it belongs to; the MCP governs nodes based on its assigned node role label. Nodes in an MCP have the same configuration; this means nodes can be scaled up and torn down in response to increased or decreased workloads.
+
+By default, there are two MCPs created by the cluster when it is installed: `master` and `worker`. Each default MCP has a defined configuration applied by the Machine Config Operator (MCO), which is responsible for managing MCPs and facilitating MCP upgrades. You can create additional MCPs, or custom pools, to manage nodes that have custom use cases that extend outside of the default node types.
+
+[NOTE]
+====
+A node can only be included in one machine config pool.
```

Comment:
We can maybe also add: custom pools take priority over worker pool

Reply:
@yuqi-zhang can you clarify what you mean by custom pools **take priority** over worker pools? What is prioritized?

Reply:
Sorry, I just meant if a node has worker,infra, it would go to the infra pool not the worker pool. Any custom pools has priority on selecting based on node labels, and any node that doesn't belong to a custom pool goes into worker